providers/redfish/firmware.go

```diff
@@ -127,8 +130,26 @@ func (c *Conn) FirmwareInstall(ctx context.Context, component, applyAt string, f
 
 	// The response contains a location header pointing to the task URI
 	// Location: /redfish/v1/TaskService/Tasks/JID_467696020275
-	if strings.Contains(resp.Header.Get("Location"), "JID_") {
-		taskID = strings.Split(resp.Header.Get("Location"), "JID_")[1]
+	var location = resp.Header.Get("Location")
+
+	taskID, err = TaskIDFromLocationURI(location)
+
+	return taskID, err
+}
+
+func TaskIDFromLocationURI(uri string) (taskID string, err error) {
+
+	if strings.Contains(uri, "JID_") {
+		taskID = strings.Split(uri, "JID_")[1]
+	} else if strings.Contains(uri, "/Monitor") {
+		// OpenBMC returns a monitor URL in Location
+		// Location: /redfish/v1/TaskService/Tasks/12/Monitor
+		splits := strings.Split(uri, "/")
+		taskID = splits[5]
```

Comment:
Its better to use regexp here or check the len of splits to make sure this doesn't panic,
```
splits := strings.Split(uri, "/")
if len(splits) == 6 {
    taskID = splits[5]
} else {
    return "", error...    
}
```

Reply:
Totally true